File: public/tile.js
const TileNone = 0
const TileGrass = 1
const TilePlankFloor = 2

const AMBIENT_LOW = 100
const AMBIENT_HALF = 175
const AMBIENT_FULL = 255

const TileLookup = new Map()
const TileTexture = []
const TileClosed = []

class Tile {
    constructor() {
        this.type = TileNone
        this.red = 0
        this.green = 0
        this.blue = 0
    }
    static Ambient(side1, side2, corner) {
        if (side1 && side2)
            return AMBIENT_LOW
        if (side1 || side2 || corner)
            return AMBIENT_HALF
        return AMBIENT_FULL
    }
}

File: client/tile.go
package main

// Constants
const (
	TileNone = 0

	AmbientLow  = 100.0
	AmbientHalf = 175.0
	AmbientFull = 255.0
)

// Variables
var (
	TileLookup  = map[string]int{}
	TileTexture [][4]float32
	TileClosed  []bool
)

// TileAmbient func
func TileAmbient(side1, side2, corner bool) float32 {
	if side1 && side2 {
		return AmbientLow
	} else if side1 || side2 || corner {
		return AmbientHalf
	}
	return AmbientFull
}

type tile struct {
	typeOf int
	red    uint8
	green  uint8
	blue   uint8
}

func tileInit() *tile {
	return &tile{}
}
